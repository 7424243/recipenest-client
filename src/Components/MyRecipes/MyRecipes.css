/* Classes */

.add-button {
    background-color: #5F7466;
    border-radius: 50%;
    border: none;
    height: 50px;
    width: 50px;
    text-align: center;
    font-size: 35px;
    float: right;
    margin-bottom: 25px;
    margin-top: 50px;
    margin-right: 50px;
    bottom: 0;
    color: white;
}

.add-button:hover {
    box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);
}

.my-recipes-list {
    list-style-type: none;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;
    list-style-type: none;
}