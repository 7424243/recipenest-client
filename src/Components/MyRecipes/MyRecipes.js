import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import config from '../../config'
import RecipenestContext from '../../RecipenestContext'
import TokenService from '../../services/token-service'
import RecipeCard from '../RecipeCard/RecipeCard'
import './MyRecipes.css'

class MyRecipes extends Component {

    //set default state
    state = {
        recipes: [],
        error: null
    }

    //allow access to content
    static contextType = RecipenestContext

    //get all of the user's recipes when the component mounts
    componentDidMount() {
        const user_id = TokenService.getUserIdFromToken()
        fetch(`${config.API_ENDPOINT}/recipes/users/${user_id}`, {
            method: 'GET',
            headers: {
                'content-type': 'application/json',
                'Authorization': `bearer ${TokenService.getAuthToken()}`
            }
        })
            .then(res => {
                if(!res.ok) {
                    return res.json().then(e => Promise.reject(e))
                }
                return res.json()
            })
            .then(recipes => {
                this.setState({recipes})
            })
            .catch(error => {
                this.setState({error})
                console.error({error})
            })
    }


    render() {
        const {recipes} = this.state
        const recipeList = recipes.length === 0 ? <p>It looks like you haven't saved any recipe notes yet! Please enter a recipe note by clicking on the '+' button to get started!</p> : recipes.map(recipe => 
            <li key={recipe.id}>
                <RecipeCard 
                    id={recipe.id}
                    name={recipe.recipe_name}
                    img_url={recipe.img_url}
                />
            </li>
        )

        return (
            <>
                <h3>My Recipes</h3>
                 <ul className='my-recipes-list'>
                     {recipeList}
                 </ul>
                <div className='container'>
                    <Link to={'/addRecipe'}><button className='add-button'>+</button></Link>
                </div>
            </>
        )
    }
}

export default MyRecipes