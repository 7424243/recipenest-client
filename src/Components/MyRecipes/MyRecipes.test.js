import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import MyRecipes from './MyRecipes';



describe.only('MyRecipes Component', () => {
    
  //smoke test
  it('renders without crashing', () => {
    Storage.prototype.getItem = jest.fn(() => 'bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoyLCJpYXQiOjE2MTE5NTYwMDMsInN1YiI6ImRlbW8ifQ.f9WIVijmP-pud7JsTd2ysx4eqtpm6JnzOkUeGRihdVE')
    
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><MyRecipes/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  //snapshot test
  it('renders the UI as expected', () => {
    Storage.prototype.getItem = jest.fn(() => 'bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoyLCJpYXQiOjE2MTE5NTYwMDMsInN1YiI6ImRlbW8ifQ.f9WIVijmP-pud7JsTd2ysx4eqtpm6JnzOkUeGRihdVE')
    const tree = renderer 
      .create(<BrowserRouter><MyRecipes/></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  
})