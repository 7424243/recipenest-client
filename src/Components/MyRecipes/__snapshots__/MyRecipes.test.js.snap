// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipes Component renders the UI as expected 1`] = `
Array [
  <h3>
    My Recipes
  </h3>,
  <ul
    className="my-recipes-list"
  />,
  <div
    className="container"
  >
    <a
      href="/addRecipe"
      onClick={[Function]}
    >
      <button
        className="add-button"
      >
        +
      </button>
    </a>
  </div>,
]
`;
